go/cli/internal/agent/frameworks/common/base_generator.go

```diff
@@ -20,90 +17,49 @@ type AgentConfig struct {
 	Framework     string
 	Language      string
 	KagentVersion string
+	McpServers    []McpServerType
+	EnvVars       []string
 }
 
-// BaseGenerator provides common functionality for all project generators
-type BaseGenerator struct {
-	TemplateFiles fs.FS
+// Implement ProjectConfig interface for AgentConfig
+func (c AgentConfig) GetDirectory() string {
+	return c.Directory
 }
 
-// NewBaseGenerator creates a new base generator
-func NewBaseGenerator(templateFiles fs.FS) *BaseGenerator {
-	return &BaseGenerator{
-		TemplateFiles: templateFiles,
-	}
+func (c AgentConfig) IsVerbose() bool {
+	return c.Verbose
 }
 
-// GenerateProject generates a new project using the provided templates
-func (g *BaseGenerator) GenerateProject(config AgentConfig) error {
-	// Get templates subdirectory
-	templateRoot, err := fs.Sub(g.TemplateFiles, "templates")
-	if err != nil {
-		return fmt.Errorf("failed to get templates subdirectory: %w", err)
-	}
-
-	// Walk through all template files
-	err = fs.WalkDir(templateRoot, ".", func(path string, d fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
-
-		// Skip directories, we'll create them as needed
-		if d.IsDir() {
-			return nil
-		}
-
-		// Determine destination path by removing .tmpl extension
-		destPath := filepath.Join(config.Directory, strings.TrimSuffix(path, ".tmpl"))
-
-		// Create the directory if it doesn't exist
-		if err := os.MkdirAll(filepath.Dir(destPath), 0755); err != nil {
-			return fmt.Errorf("failed to create directory %s: %w", filepath.Dir(destPath), err)
-		}
-
-		// Read template file
-		templateContent, err := fs.ReadFile(templateRoot, path)
-		if err != nil {
-			return fmt.Errorf("failed to read template file %s: %w", path, err)
-		}
-
-		// Render template content
-		renderedContent, err := g.renderTemplate(string(templateContent), config)
-		if err != nil {
-			return fmt.Errorf("failed to render template for %s: %w", path, err)
-		}
-
-		// Create file
-		if err := os.WriteFile(destPath, []byte(renderedContent), 0644); err != nil {
-			return fmt.Errorf("failed to write file %s: %w", destPath, err)
-		}
-
-		if config.Verbose {
-			// print the generated files
-			fmt.Printf("  Generated: %s\n", destPath)
-		}
-
-		return nil
-	})
+func (c AgentConfig) ShouldInitGit() bool {
+	return false // Agent projects don't auto-init git
```

Comment:
Why not?

Reply:
we didn't do it before, nothing preventing us from doing it though